support/util/cpocheck.go

```diff
@@ -0,0 +1,13 @@
+package util
+
+import hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+
+// IsCPOCompatibleWithHCP checks to see if the active image in a control-plane-operator component matches the
+// expected image listed in the HCP object
+func IsCPOCompatibleWithHCP(hcpAnnotations map[string]string, activeImage string) bool {
+	expectedImage, ok := hcpAnnotations[hyperv1.ControlPlaneOperatorImageAnnotation]
+	if !ok {
+		return false
+	}
+	return expectedImage == activeImage
```

Comment:
Nit: If it doesn't exist, `expectedImage` is an empty string, so you can just do
```suggestion
	return hcpAnnotations[hyperv1.ControlPlaneOperatorImageAnnotation] == activeIage
```

Reply:
done